pandas/io/parsers.py

```diff
@@ -2165,6 +2231,129 @@ def _maybe_parse_dates(self, values, index, try_parse_dates=True):
+
+        self.usecols, self.usecols_dtype = _validate_usecols_arg(self.kwds["usecols"])
+        na_values = self.kwds["na_values"]
+        if isinstance(na_values, dict):
+            raise ValueError(
+                "The pyarrow engine doesn't support passing a dict for na_values"
+            )
+        self.na_values = list(
+            _clean_na_values(
+                self.kwds["na_values"], keep_default_na=self.kwds["keep_default_na"]
+            )[0]
+        )
+
+        if isinstance(self.src, TextIOBase):
+            self.src = BytesIOWrapper(self.src, encoding=self.encoding)
+
+    def _get_pyarrow_options(self):
+        # rename some arguments to pass to pyarrow
+        mapping = {
+            "usecols": "include_columns",
+            "na_values": "null_values",
+            "escapechar": "escape_char",
+            "skip_blank_lines": "ignore_empty_lines",
+        }
+        for pandas_name, pyarrow_name in mapping.items():
+            if pandas_name in self.kwds and self.kwds.get(pandas_name) is not None:
+                self.kwds[pyarrow_name] = self.kwds.pop(pandas_name)
+
+        self.parse_options = {
+            option_name: option_value
+            for option_name, option_value in self.kwds.items()
+            if option_value is not None
+            and option_name
+            in ("delimiter", "quote_char", "escape_char", "ignore_empty_lines")
+        }
+        self.convert_options = {
+            option_name: option_value
+            for option_name, option_value in self.kwds.items()
+            if option_value is not None
+            and option_name
+            in ("include_columns", "null_values", "true_values", "false_values")
+        }
+        self.read_options = {
+            "autogenerate_column_names": self.header is None,
+            "skip_rows": self.header
+            if self.header is not None
+            else self.kwds["skiprows"],
+        }
+
+    def _finalize_output(self, frame):
+        num_cols = len(frame.columns)
+        if self.header is None:
+            if self.names is None:
+                if self.prefix is not None:
+                    self.names = [f"{self.prefix}{i}" for i in range(num_cols)]
+                elif self.header is None:
+                    self.names = range(num_cols)
+            frame.columns = self.names
+        # we only need the frame not the names
+        frame.columns, frame = self._do_date_conversions(frame.columns, frame)
+        if self.index_col is not None:
+            for i, item in enumerate(self.index_col):
+                if is_integer(item):
+                    self.index_col[i] = frame.columns[item]
+            frame.set_index(self.index_col, drop=True, inplace=True)
+
+        if self.kwds.get("dtype") is not None:
+            frame = frame.astype(self.kwds.get("dtype"))
+        return frame
+
+    def read(self):
+        pyarrow = import_optional_dependency("pyarrow.csv")
```

Comment:
I am fairly certain that version checking is broken for submodule imports like this, which is part of the reason I was modifying import_optional_dependency. Can you make sure that this code errors with pyarrow < 0.15?

Reply:
I can (but also that sounds like something that should have its own issue and a separate PR)

Reply:
Ok, I'll fix this as a followup

Reply:
I pulled your fix from the old branch here and it works
https://github.com/pandas-dev/pandas/compare/master...arw2019:submodule-import-optional-dep?expand=1
the problem is the test (I haven't quite figured out how to add a mock submodule I think)

I'd say that's orthogonal to this PR so can do whenever

Reply:
yeah just make an issue for followups with checkboxes is fine

Reply:
different than #38872 as that is implementing features, these are more folllowups about style / testing and so on

Reply:
> yeah just make an issue for followups with checkboxes is fine

#38889

Reply:
> Ok, I'll fix this as a followup

tracking at #38888

Reply:
Will be fixed by #38925